### ISSUE TITLE
plugin-auth-backend: Okta auth adding base path twice in redirect uri

### ISSUE THREAD

**Issue by sachin-kumar-ik8** [backstage/backstage#8411]
When we configure service discovery then auth callback url include backend baseurl base path + service discovery base path. 

## Expected Behavior

Redirect uri shouldnt include base path twice. It should be for ex:
redirect_uri=http://localhost:7000/backstage-api/auth/okta/handler/frame

## Current Behavior
When we add backend baseUrl with base path in config.yaml and set ServiceDiscovery config option with basePath then okta rediect uri looks like:
redirect_uri=http://localhost:7000/backstage-api/backstage-api/auth/okta/handler/frame

app-config.yaml:
    
```
       backend:
         baseUrl: http://localhost:7000/backstage-api
```
    

packages/backend/src/index.ts:
```
  const discovery = SingleHostDiscovery.fromConfig(config, { 
    basePath: "/backstage-api"
  });
  ```

## Possible Solution

plugin-auth-backend: dist/index.cjs.js -> inside createOktaProvider function callbackurl is
http://localhost:7000/backstage-api/backstage-api/auth/okta/handler/frame

## Steps to Reproduce

1. Add backend baseUrl with basepath something other than : http://localhost:7000/backstage-api
2. Add service discovery with config and basepath option
```
  const discovery = SingleHostDiscovery.fromConfig(config, { 
    basePath: "/backstage-api"
  });
  ```
3. Start express server with basepath in router
```
  const service = createServiceBuilder(module)
    .loadConfig(config)
    .addRouter('/backstage-api', apiRouter);
    ```

## Context

Okta redirect uri is failing with 400

Environment: development
app-config.yaml
```
app:
  title: backstage
  baseUrl: http://localhost:3000/backstage/

backend:
  baseUrl: http://localhost:7000/backstage-api
...

auth:
  environment: development
  providers:
    okta:
      development:
        clientId: test
        clientSecret: test
        audience: https://dev-123354.okta.com
...
...
     ```

- NodeJS Version (v14):
- Operating System and Version (e.g. Ubuntu 14.04): macos
- Browser Information:

**Comment by Rugvip** [comment]
@sachin-kumar-ik8 The `backend.baseUrl` config should point to the publicly reachable root URL of the backend, which while I see why the confusion exists, does not include the `/backstage-api` path. Another way to look at it is that the `backend.baseUrl` should point to the service created with `createServiceBuilder`, which would just be `http://localhost:7000`, since `/backstage-api` is a route added to the service builder.

So TL;DR: switch `backend.baseUrl` to `http://localhost:7000` and it should work. I'm expecting that to solve it so I'll close the issue, but feel free to reopen it if there's any trouble!

**Comment by sachin-kumar-ik8** [comment]
@Rugvip when I switch `backend.baseUrl` to` http://localhost:7000 ` then frontend is trying to reach `http://localhost:7000 /api/<api-name>` which doesnt exist and crash with 404. Because my server is running with base path 

` const service = createServiceBuilder(module)
    .loadConfig(config)
    .addRouter('/backstage-api', apiRouter);`.

**Comment by Rugvip** [comment]
@sachin-kumar-ik8 Ah alright, that'd be a frontend discovery issue then. Will say that overriding the `/api` path is a somewhat advanced use-case, but supported. You'll want to override the `DiscoveryApi` implementation in the frontend with one that uses the matching pattern. Basically add a custom `createApiFactory` call to `packages/app/src/apis.ts` that matches the [default one](https://github.com/backstage/backstage/blob/a3ec122170e0205fd3f9c307b98b1c5e4f55bf5f/packages/app-defaults/src/defaults/apis.ts#L68) but using an updated pattern.
